### ISSUE TITLE
Heart beat management from RP to QCA

### ISSUE THREAD

**Issue by mturilli** [radical-collaboration/QCArchive#12]
To be a requirement.

**Comment by mturilli** [comment]
REST interface on RP-side is not required anymore

**Comment by dgasmith** [comment]
We will still need heartbeats from RP to double check that it is still alive. This is a work in progress in QCA, please give us a few weeks to get this properly implemented.

**Comment by andre-merzky** [comment]
I assume that you'll define a REST method you want us to call on regular intervals?   Well, we'll be pulling for tasks at regular intervals, that could naively be used as a heartbeat, too.  But I also don't mind doing that cleanly and calling a separate REST method, of course.

**Comment by dgasmith** [comment]
Yes to a heartbeat REST method.

We could set it up as a task pull counts as a heartbeat as well. Do you have thoughts/experience convolving heartbeats/task pulls together? Technically easy to do, but having a clean separation does have upsides.

**Comment by andre-merzky** [comment]
Yeah, I don't mind clean separation - we do the same in RP.  Not for semantic reasons though, but for performance and reliability: I don't want to have heartbeats stuck in large task queues...  But that's only the case because we intermediate via a DB right now, and cannot evaluate the actual connection as heartbeat...

**Comment by mturilli** [comment]
@dgasmith to implement the dedicated interface in at least 1 month. Meanwhile QCA will do manual boo-keeping of suspended jobs follow a RP failure.

**Comment by mturilli** [comment]
Ongoing
